From 1eb7f9e64fdaf71a84e5403932ade0d9d3565e8d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 17:20:33 +0800
Subject: [PATCH 267/301] AOSCOS: staging: sb105x: revise type of
 mp_write_room() as unsigned int

Per commit 03b3b1a2405c ("tty: make tty_operations::write_room return
uint"), member function `write_room()' in `struct tty_operations' was
revised from type `int' to type `unsigned int'.

Revise driver code accordingly.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index f61451fabdcd..593835f0ad11 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -40,7 +40,7 @@ static int mp_put_char(struct tty_struct *tty, unsigned char ch);
 
 static void mp_put_chars(struct tty_struct *tty);
 static ssize_t mp_write(struct tty_struct *tty, const unsigned char *buf, long unsigned int count);
-static int mp_write_room(struct tty_struct *tty);
+static unsigned int mp_write_room(struct tty_struct *tty);
 static int mp_chars_in_buffer(struct tty_struct *tty);
 static void mp_flush_buffer(struct tty_struct *tty);
 static void mp_send_xchar(struct tty_struct *tty, char ch);
@@ -671,7 +671,7 @@ static ssize_t mp_write(struct tty_struct *tty, const unsigned char *buf, long u
 	return ret;
 }
 
-static int mp_write_room(struct tty_struct *tty)
+static unsigned int mp_write_room(struct tty_struct *tty)
 {
 	struct sb_uart_state *state = tty->driver_data;
 
-- 
2.48.1

